akka-remote/src/main/scala/akka/remote/artery/ArterySettings.scala

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import akka.stream.ActorMaterializerSettings
+import akka.util.Helpers.{ ConfigOps, Requiring }
+import akka.util.WildcardIndex
+import akka.NotUsed
+import com.typesafe.config.Config
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+
+import java.net.InetAddress
+import java.util.concurrent.TimeUnit
+
+/** INTERNAL API */
+private[akka] final class ArterySettings private (config: Config) {
+  import config._
+  import ArterySettings._
+
+  val Enabled: Boolean = getBoolean("enabled")
+  val Port: Int = getInt("port")
+  val Hostname: String = getString("hostname") match {
+    case "" | "<getHostAddress>" ⇒ InetAddress.getLocalHost.getHostAddress
+    case "<getHostName>"         ⇒ InetAddress.getLocalHost.getHostName
+    case other                   ⇒ other
+  }
+  val LargeMessageDestinations =
+    config.getStringList("large-message-destinations").asScala.foldLeft(WildcardIndex[NotUsed]()) { (tree, entry) ⇒
+      val segments = entry.split('/').tail
+      tree.insert(segments, NotUsed)
+    }
+
+  object Advanced {
```

Comment:
doesn't this style make it impossible to programatically change values?

See https://github.com/ktoso/kaze-class style and how HTTP deals with settings perhaps?